platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SignedState.java

```diff
@@ -264,11 +265,13 @@ public void setSigSet(@NonNull final SigSet sigSet) {
      * {@inheritDoc}
      */
     @Override
-    public @NonNull AddressBook getAddressBook() {
-        return Objects.requireNonNull(
-                RosterUtils.buildAddressBook(RosterRetriever.retrieveActiveOrGenesisRoster(
-                        (MerkleStateRoot) getState().getSwirldState())),
-                "address book stored in this signed state is null, this should never happen");
+    public Roster getRoster() {
+        /*
+        Ideally the roster would be captured in the constructor but due to the mutable underlying state, the roster
+        can change from underneath us. Therefore, the roster must be regenerated on each access.
+         */
+        return RosterRetriever.retrieveActiveOrGenesisRoster(
+                (MerkleStateRoot) getState().getSwirldState());
```

Comment:
I know that the `retrieveActiveOrGenesisRoster` may return null. However, previously this method guaranteed that the return value is non-null. I believe we may have code that relies on this. And even if we don't, I believe a `SignedState` should always have a non-null Roster - otherwise it couldn't have been signed in the first place because signatures are supposed to come from the nodes of the network, and the roster specifies the nodes. So if there's no nodes, there cannot be signatures, and hence the state cannot be signed. In other words, while `retrieveActiveOrGenesisRoster` could return null in theory, it should never return `null` when called on a (real, or properly mocked) SignedState.

Could we restore the `@NonNull` property of this method?